modal/io_streams.py

```diff
@@ -124,6 +128,54 @@ async def read(self) -> str:
 
         return data
 
+    async def _consume_container_process_stream(self):
+        """mdmd:hidden
+        Consumes the container process stream and stores the messages in the buffer.
+        """
+        completed = False
+        retries_remaining = 10
+        while not completed:
+            try:
+                iterator = _container_process_logs_iterator(self._object_id, self._file_descriptor, self._client)
+
+                async for message in iterator:
+                    self._container_process_buffer.append(message)
+                    if message is None:
+                        completed = True
+                        break
+
+            except (GRPCError, StreamTerminatedError) as exc:
```

Comment:
I don't love the retry code duplication here, but I couldn't figure out how to get rid of it. I almost got a helper function working, but the signature didn't work out since one of these yields and the other doesn't.